### ISSUE TITLE
Negative memory increment (memory decrement) is displayed incorrectly

### ISSUE THREAD

**Issue by espdev** [pythonprofilers/memory_profiler#226]
We use:
* Python 3.7.1
* memory-profiler 0.55.0 (from PyPI)

We run the example from readme:
```python
from memory_profiler import profile


@profile
def my_func():
    a = [1] * (10 ** 6)
    b = [2] * (2 * 10 ** 7)
    del b
    return a

if __name__ == '__main__':
    my_func()
```

And get the result:
```
Line #    Mem usage    Increment   Line Contents
================================================
     4     35.7 MiB     35.7 MiB   @profile
     5                             def my_func():
     6     43.1 MiB      7.4 MiB       a = [1] * (10 ** 6)
     7    195.8 MiB    152.6 MiB       b = [2] * (2 * 10 ** 7)
     8     43.3 MiB      0.0 MiB       del b   # <--- expected  -152.5 MiB
     9     43.3 MiB      0.0 MiB       return a
```

We expect to see -152.5 MiB "Increment" but we see 0.0 MiB. We also see that "Mem usage" value is correct.

**Comment by jitseniesen** [comment]
Reverting commit 5fe38da926738a7715c860053a498859e2db0c87 fixes this issue.

**Comment by mcepl** [comment]
Yes, I can confirm that [reverting](https://build.opensuse.org/package/view_file/devel:languages:python/python-memory_profiler/revert-5fe38da92673.patch?expand=1)  5fe38da926738a7715c860053a498859e2db0c87 makes the test suite of spyder-memory-profiler pass (thus fixing https://github.com/spyder-ide/spyder-memory-profiler/issues/20 ).

**Comment by felicitywang** [comment]
@jitseniesen Thank you for the great work. I installed commit 5fe38da as suggested but I'm still seeing the same result shown above. Any suggestions? Thank you!

**Comment by jitseniesen** [comment]
@felicitywang I do not understand you. Commit 5fe38da is part of version 0.55 and later, so I don't understand what you mean when you say that you installed that commit. In fact, I said that you should revert that commit to fix the issue.

**Comment by rongjiang9037** [comment]
Hi,

I'm new to python.. I have the same issue with version 0.57.0, python 3.5.3.

If I install the package with "pip", how can I revert the commit?
Thanks!!!!
Best,
Rong

**Comment by zaccharieramzi** [comment]
I also had this issue. Since the commit @jitseniesen is referring to is dated from 28 Jun 2018, the version that can work is 0.52 (I guess installing with a given version is simpler than installing from source with a given commit).

So the command would be `pip install -U memory-profiler==0.52`. This worked for me (as in I managed to see negative increments).

However, the results I am seeing are very weird...

```
Filename: /content/mprun_demo.py

Line #    Mem usage    Increment   Line Contents
================================================
     1    151.5 MiB    151.5 MiB   def sum_of_lists(N):
     2    151.5 MiB      0.0 MiB       total = 0
     3    168.6 MiB      0.0 MiB       for i in range(5):
     4    199.3 MiB -46768119.2 MiB           L = [j ^ (j >> i) for j in range(N)]
     5    199.3 MiB      0.0 MiB           total += sum(L)
     6    168.6 MiB   -123.0 MiB           del L # remove reference to L
     7    168.6 MiB      0.0 MiB       return total
```

I am trying to go back through the versions to see if I can get a working one. @fabianp do you have an idea of a version that might be working for this simple example found [here](https://jakevdp.github.io/PythonDataScienceHandbook/01.07-timing-and-profiling.html)?

**Comment by zaccharieramzi** [comment]
0.41 seems to be working fine with this example.

**Comment by fabianp** [comment]
I don't know of a fix, but I would be more than happy to review pull requests that fix this behavior

**Comment by piyh** [comment]
Still getting this and it looks really bizarre. 

`from pathlib import Path
from memory_profiler import profile


@profile
def stripSpecialChar():
'''
Filename: C:/Users/ryan.mcgrath/Desktop/read-write line by line.py

Line #    Mem usage    Increment  Occurences   Line Contents
============================================================
     5     34.2 MiB     34.2 MiB           1   @profile
     6                                         def stripSpecialChar():
     7                                                 
     8     34.2 MiB      0.0 MiB           1       inFile = Path('C:/Users/ryan.mcgrath/Desktop/Files/Insider Flag/BFSInternalInterfaces_20200701 - Copy.txt')
     9     34.2 MiB      0.0 MiB           1       outFile = Path('C:/Users/ryan.mcgrath/Desktop/Files/Insider Flag/BFSInternalInterfaces_20200701.txt2')
    10                                         
    11     34.2 MiB      0.0 MiB           1       with open(inFile,'r') as r:
    12     34.2 MiB      0.0 MiB           1           with open (outFile, 'w') as w:
    13                                                     #x = r.read()
    14                                                     #x = x.replace('\\','') #strip out any backslash
    15                                                     #w.write(x)
    16                                         
    17     34.3 MiB  -4847.4 MiB      300769               for l in r:
    18     34.3 MiB  -4847.3 MiB      300768                   w.write(l.replace('|','UwU'))
'''
    inFile = Path('C:/Users/ryan.mcgrath/Desktop/Files/Insider Flag/BFSInternalInterfaces_20200701 - Copy.txt')
    outFile = Path('C:/Users/ryan.mcgrath/Desktop/Files/Insider Flag/BFSInternalInterfaces_20200701.txt2')

    with open(inFile,'r') as r:
        with open (outFile, 'w') as w:
            #x = r.read()
            #x = x.replace('\\','') #strip out any backslash
            #w.write(x)

            for l in r:
                w.write(l.replace('|','UwU'))
                
if __name__ == '__main__':
    stripSpecialChar()

`

**Comment by samanemami** [comment]
I think the most stable version is `0.52`.

**Comment by ilaflott** [comment]
On RHEL8, with conda-forge Python 3.9.15, `memory_profiler` 0.61.0, and `psutil` 5.9.1, I saw the massive non-sensible negative increments in my output as well, doing `python -m memory_profiler <my_for_loop_abuse>.py`.

After reading the `README.rst`, and stumbling upon the warning [in this section](https://github.com/pythonprofilers/memory_profiler/blob/master/README.rst#time-based-memory-usage),r I had a vague hunch that I shouldn't keep `from psutil import Process` in my script (I had tried to use this before learning about this module).

Upon removing the import statement and commenting out the relevant lines downstream, my increments returned to normal.

can provide more info/detail if anyone who tries to maintain this would find it helpful. Otherwise, avoid importing other profiling tools in your script while trying to use `memory_profiler`.

**Comment by glaubitz** [comment]
Is reverting this patch still necessary?
